### PR TITLE
Faster zero shot image

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Should superseed: https://github.com/huggingface/transformers/pull/21861#event-8644623666

- Keeps `batch_size` active, but restricts it to the `image`part.
- Calculates `candidate_labels` features only once (there's no way with this current pipeline to send different candidate_labels so we can take the optimization.
- Ultra narrowed for `CLIP` but the pipeline has existed for 4 months now with no new models. Given the popularity of CLIP for diffusion models, I think this is ok to overspecify. We can always relax later when new models come in.

This allows to downgrade from ChunkPipeline to regular PIpeline

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Narsil** [comment]
@yessenzhar

File: src/transformers/pipelines/zero_shot_image_classification.py [hunk 1]
```diff
@@ -107,42 +107,37 @@ def _sanitize_parameters(self, **kwargs):
...
-            sequence = hypothesis_template.format(candidate_label)
-            inputs = self.tokenizer(sequence, return_tensors=self.framework)
-            inputs["pixel_values"] = images.pixel_values
-            yield {"is_last": i == n - 1, "candidate_label": candidate_label, **inputs}
+        image = load_image(image)
+        inputs = self.image_processor(images=[image], return_tensors=self.framework)
+        inputs["candidate_labels"] = candidate_labels
+        sequences = [hypothesis_template.format(x) for x in candidate_labels]
+        text_inputs = self.tokenizer(sequences, return_tensors=self.framework, padding=True)
+        inputs["text_inputs"] = [text_inputs]
```
Comment (Narsil) [inline src/transformers/pipelines/zero_shot_image_classification.py:115]: We're doing this to prevent batching at the `input_ids` level, since we're going to take only 1 in the end, even in the batching case.

**Review by gante — APPROVED** [review]
LGTM 👍

**Review by sgugger — APPROVED** [review]
Thanks, LGTM!

**Review by ArthurZucker — APPROVED** [review]
looks good to me. This can also be applied to the zero shot audio classification pipeline no?

**Comment by Narsil** [comment]
Does it also do a cross product ? Then possibly yes.

**Comment by ArthurZucker** [comment]
I mostly copy pasted the whole sequence. Will open a PR

File: src/transformers/pipelines/zero_shot_image_classification.py [hunk 2]
```diff
@@ -17,18 +18,16 @@
     from ..image_utils import load_image
 
 if is_torch_available():
-    import torch
+    pass
```
Comment (ArthurZucker) [inline src/transformers/pipelines/zero_shot_image_classification.py:21]: to remove?

Comment (Narsil) [reply src/transformers/pipelines/zero_shot_image_classification.py:21]: Actually I think we can remove thanks to @yessenzhar

**Comment by ArthurZucker** [comment]
CLAP has the same implementation for loss and etc. So we should see a similar behaviour

**Comment by yessenzhar** [comment]
@Narsil Thank you for sorting this out this quick.
